mlrun/datastore/dbfs_store.py

```diff
@@ -0,0 +1,146 @@
+                "Invalid key parameter - key must start with '/'"
+            )
+        return key
+
+    def get(self, key: str, size=None, offset=0) -> bytes:
+        key = self._prepare_path_and_verify_filesystem(key)
+        if size is not None and size <= 0:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "size cannot be negative or zero"
+            )
+        if not size and not offset:
+            return self._filesystem.cat_file(key)
+            # the maximum number of allowed bytes to read is 1MB
+        end = offset + (size or ONE_MB)
+        return self._filesystem._get_data(key, start=offset, end=end)
+
+    def put(self, key, data, append=False):
+
+        key = self._prepare_path_and_verify_filesystem(key)
+        if append:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Append mode not supported for Databricks file system"
+            )
+        #  can not use append mode because it overrides data.
+        mode = "w"
+        if isinstance(data, bytes):
+            mode += "b"
+        elif not isinstance(data, str):
+            raise TypeError(f"Unknown data type {type(data)}")
+        with self._filesystem.open(key, mode) as f:
+            f.write(data)
+
+    def upload(self, key: str, src_path: str):
+        key = self._prepare_path_and_verify_filesystem(key)
+        self._filesystem.put_file(src_path, key, overwrite=True)
+
+    def stat(self, key: str):
+        key = self._prepare_path_and_verify_filesystem(key)
+        files = self._filesystem.ls(key, detail=True)
+        if len(files) == 1 and files[0]["type"] == "file":
+            size = files[0]["size"]
+        elif len(files) == 1 and files[0]["type"] == "directory":
+            raise FileNotFoundError("Operation expects a file not a directory!")
+        else:
+            raise ValueError("Operation expects to receive a single file!")
+        return FileStats(size, None)
+
+    def listdir(self, key: str):
+        """
+        Basic ls of file/dir - without recursion.
+        """
+        key = self._prepare_path_and_verify_filesystem(key)
+        if self._filesystem.isfile(key):
+            return key
+        remote_path = f"{key}/*"
+        files = self._filesystem.glob(remote_path)
+        key_length = len(key)
+        #  Get only the files and directories under key path, without the key path itself.
+        # for example for /test_mlrun_dbfs_objects/test.txt the function will return ['test.txt'].
+        files = [
+            file.split("/", 1)[1][key_length:]
+            for file in files
+            if len(file.split("/")) > 1
+        ]
+        return files
+
+    def supports_isdir(self):
+        return False
+
+    def rm(self, path, recursive=False, maxdepth=None):
+        if maxdepth:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "dbfs file system does not support maxdepth option in rm function."
```

Comment:
```suggestion
                "dbfs file system does not support maxdepth option in rm function"
```
(code base convention)